extensions/vscode/src/state.ts

```diff
@@ -191,14 +203,27 @@ export class PublisherState implements Disposable {
   }
 
   async refreshContentRecords() {
-    const api = await useApi();
-    const response = await api.contentRecords.getAll(".", { recursive: true });
+    try {
+      await showProgress("Refreshing Deployments", Views.HomeView, async () => {
```

Comment:
I'm not adamantly against it, but I do find it strange that the Home View doesn't control its own `showProgress` and instead the responsibility gets passed here.

Perhaps we could do something more generic on the `homeView` side?

Reply:
I understand that it could feel strange that the Home View doesn't control the show progress, but at the same time that might be good? 

I think it could be nice that every place using `PublisherState` does not have to remember to wrap methods in `showProgress`.

On the other hand, while taking a look to other methods, we don't use `showProgress` consistently, e.g: `getSelectedConfiguration` does not use it, I'm tempted to remove it at all on these methods.

It is nice to have visual feedback that something is happening but these operations do not seem to take enough time for the progress indicator to show up, unless I'm missing some background history on this. What do you folks think if we remove `showProgress` at all from these methods?

Reply:
> I understand that it could feel strange that the Home View doesn't control the show progress, but at the same time that might be good?
I think it could be nice that every place using PublisherState does not have to remember to wrap methods in showProgress.

Yeah you are right. I always struggle a bit to place behavior like this 😓 

> What do you folks think if we remove `showProgress` at all from these methods?

These operations are very quick, but they can get slower if a user has a lot of them. I'm sure that will be very rare, if it happens at all, but in a large workspace it could. If it does occur it would be nice to have a progress bar.